tests/v1/entrypoints/openai/test_completion.py

```diff
@@ -35,26 +49,124 @@ def default_server_args():
                             "--no-enable-prefix-caching",
                             "--disable-frontend-multiprocessing"
                         ]])
-def server(default_server_args, request):
+def args(default_server_args, request):
     if request.param:
         default_server_args.extend(request.param)
-    with RemoteOpenAIServer(MODEL_NAME, default_server_args) as remote_server:
-        yield remote_server
+    parser = FlexibleArgumentParser(description="vLLM's remote OpenAI server.")
+    parser = make_arg_parser(parser)
+    args = parser.parse_args(["--model", MODEL_NAME, *default_server_args])
+    return args
+
+
+@pytest_asyncio.fixture(scope="module")
+async def server_logic(args):
+    async with build_async_engine_client(args) as engine_client:
+        vllm_config = await engine_client.get_vllm_config()
+        model_config = vllm_config.model_config
+
+        served_model_names = args.served_model_name or [args.model]
+        base_model_paths = [
+            BaseModelPath(name=name, model_path=args.model)
+            for name in served_model_names
+        ]
+        models = OpenAIServingModels(
+            engine_client=engine_client,
+            model_config=model_config,
+            base_model_paths=base_model_paths,
+            lora_modules=args.lora_modules,
+            prompt_adapters=args.prompt_adapters,
+        )
+
+        request_logger = None if args.disable_log_requests else RequestLogger(
+            max_log_len=args.max_log_len)
+        server_logic = OpenAIServingCompletion(
+            model_config=model_config,
+            engine_client=engine_client,
+            models=models,
+            request_logger=request_logger,
+            return_tokens_as_token_ids=args.return_tokens_as_token_ids,
+        )
+
+        yield server_logic
 
 
-@pytest_asyncio.fixture
-async def client(server):
-    async with server.get_async_client() as async_client:
-        yield async_client
+@pytest_asyncio.fixture(scope="module")
+async def client():
+    yield openai.AsyncOpenAI(
+        api_key="EMPTY",
+        base_url="http://localhost:8000/v1",
+        max_retries=0,
+    )
+
 
+@pytest.fixture
+def register_completion(httpx_mock: HTTPXMock):
+    """Register a callback for the completion endpoint."""
 
-@pytest.mark.asyncio
+    def _register(handler):
+        httpx_mock.add_callback(
+            handler,
+            url="http://localhost:8000/v1/completions",
+            method="POST",
+        )
+
+    return _register
+
+
+def make_handle_request_no_streaming(server_logic):
```

Comment:
probably better to make this into a module hook

https://docs.pytest.org/en/7.1.x/how-to/writing_hook_functions.html